test/fixtures/es-module-loaders/hook-resolve-type-loader.mjs

```diff
@@ -0,0 +1,18 @@
+/** @type {MessagePort} */
+let port;
+export function initialize(data) {
+  port = data.port;
+}
```

Comment:
nit

```suggestion
export function initialize(data) {
  ({ port } = data);
}
```

Reply:
I don’t want to shadow the `port` variable declared higher up, I think that can be very confusing.

Reply:
O.o That's not what this does: this is a destructuring re-assignment. The `port` in the parent scope (line 2) is assigned the value of `data.port`.

Reply:
I know, but I find it confusing. It’s not as obvious (to me anyway) that it’s referencing a previously declared variable.